### DISCUSSION TITLE
Session field and pagination

### DISCUSSION THREAD

**Discussion by jrizz24** [metarank/metarank#1302] [General]
Hi There, 

I am wondering how to handle pagination when logging events to the feedback API. We have a continuous scroll SERP. Should each page request  have a unique ID in the ranking event and all have the same `session` value? Or should all ranking events have the same ID? 

The documentation doesn't really make it clear how the `session` field is used or what it should really represent. 

Thank in advance!

**Comment by shuttie** [comment]
Hi @jrizz24 

Metarank does currently not perform any custom logic on top of pagination. To cover your use-case with continuous scroll, I suggest to do the following:
* even when UI does a continuous scrolling, on the backend you still probably perform pagination in chunks
* option 1: rerank only within a single page. So when you retrieve a next page, you also rerank it before sending to customer. Simple to implement, but cannot do inter-page reranking.
* option 2: rerank multiple pages at once. You have different client page size and reranking page sizes: so when client asks for page 1, you retrieve pages 1-2-3-4, rerank them as altogether and cut the first page. You may also need to persist the ranking to maintain a consistent pagination.

As for feedback events, Metarank expects:
1. ranking/search events as a root event type. This is used to implicitly labels results later: when there was an interaction on a document, then it's considered relevant. 
2. Interaction events. When such an event is received, it's joined with the root ranking event, so you can know what was displayed to the customer, and what docs got interactions later.

So for pagination we usually expect any type of ranking event: it can be just current page (so you'll get a new event id for each page), or event for multiple pages at once. You can even sent ranking events having the same id, and the latter will overwrite the former.

Also note that `user` field is a user identifier, single user may have multiple `session`s on the site (so it's not a search session, but more like a google-analytics style session when you do multiple actions within short amount of time). And within session there can be multiple searches performed (so with different event ids).

**Reply by jrizz24** [reply]
Hi @shuttie 

This really clears things up. Thank you for your thorough answer.
